packages/dev/core/src/Meshes/meshUVSpaceRenderer.ts

```diff
@@ -145,36 +195,152 @@ export class MeshUVSpaceRenderer {
      * @param angle The rotation angle around the direction of the projection
      */
     public renderTexture(texture: BaseTexture, position: Vector3, normal: Vector3, size: Vector3, angle = 0): void {
+        // Create the diffuse render target texture if it doesn't exist
+        if (!this.texture && !this._options.uvEdgeBlending && !this._decalTexture) {
+            this._updateRTT();
+        } else if (this.texture && !this._userCreatedTextureSetup && !this._options.uvEdgeBlending && !this._decalTexture) {
+            this._updateRTT();
+        } else if (this.texture && !this._userCreatedTextureSetup && this._options.uvEdgeBlending && !this._decalTexture) {
+            this._createDecalDiffuseRTT();
+        } else if (!this.texture && this._options.uvEdgeBlending && !this._decalTexture) {
+            this._createDecalDiffuseRTT();
+        }
+
+        // // Prepare the shader with the decal texture, mask texture, and necessary uniforms
+        const shader = MeshUVSpaceRenderer._GetShader(this._scene);
+        shader.setTexture("textureSampler", texture); // Decal texture
+
+        // Calculate and set the projection matrix
+        const projectionMatrix = this._createProjectionMatrix(position, normal, size, angle);
+        shader.setMatrix("projMatrix", projectionMatrix);
+
+        if (!this._options.uvEdgeBlending) {
+            if (this.texture instanceof RenderTargetTexture) {
+                this.texture.render();
+            }
+        } else {
+            if (MeshUVSpaceRenderer._IsRenderTargetTexture(this._decalTexture)) {
+                if (this._options.uvEdgeBlending) {
+                    this._decalTexture.render();
+                    this._createMaskTexture();
+                    this._createFinalTexture();
+                }
+            }
+        }
+    }
+    /**
+     * Creates a texture RTT if one doesn't exist,
+     */
+    _updateRTT() {
         if (!this.texture) {
-            this._createDiffuseRTT();
+            this.texture = this._createRenderTargetTexture(this._options.width, this._options.height);
+        } else {
+            this._userCreatedTextureSetup = true;
+        }
+        if (this.texture instanceof RenderTargetTexture) {
+            this.texture.renderList = [this._mesh];
+            this.texture.setMaterialForRendering(this._mesh, MeshUVSpaceRenderer._GetShader(this._scene));
+            this.texture.onClearObservable.addOnce(() => {
+                this._scene.getEngine().clear(this.clearColor, true, true, true);
+                if (this.texture instanceof RenderTargetTexture) {

```

Comment:
you can remove the following as you call addOnce, the obsevable will be "cleaned up" automatically. Adding an extra empty function is only decreasing perf.

Reply:
I took this from the existing _createRenderTargetTexture function which was using it to generate the decal texture and from the comment next to it:

 // this disables clearing the texture for the next frames

Reply:
Yes, you should keep the empty function, it's the current way to avoid the RTT to be automatically cleared (we should probably add a property for this one day!).